mono/metadata/object.c

```diff
@@ -5305,44 +5312,46 @@ mono_runtime_try_invoke_array (MonoMethod *method, void *obj, MonoArray *params,
 	}
 }
 
-/**
- * mono_object_new:
- * \param klass the class of the object that we want to create
- * \returns a newly created object whose definition is
- * looked up using \p klass.   This will not invoke any constructors, 
- * so the consumer of this routine has to invoke any constructors on
- * its own to initialize the object.
- * 
- * It returns NULL on failure.
- */
+// FIXME these will move to header soon
+static  MonoObjectHandle
+mono_object_new_alloc_by_vtable (MonoVTable *vtable, MonoError *error);
+
+// FIXME these will move to header soon
+static MonoObjectHandle
+mono_object_new_by_vtable (MonoVTable *vtable, MonoError *error);
+
 MonoObject *
 mono_object_new (MonoDomain *domain, MonoClass *klass)
```

Comment:
I think @luhenry tried to implement `mono_object_new` in terms of `mono_object_new_handle` about two years ago.  I don't recall what the feedback was for why we couldn't merge it at the time.  @luhenry do you remember?

Reply:
I'm ambivalent on this. Either way helps port the next layers up and we can fill in cracks later. There are 100+ users of the raw pointer allocators (just searching for object_new, not even foo_new) My larger concern was if I was duplicating anything that already exists. Here for a change C++ wins at plain text search as they'd all be same-named overloads and I'd find them all automatically. With the C approach I have to hope I chose the same name as you.